cnudie/util.py

```diff
@@ -470,3 +472,89 @@ def enumerate_group_pairs(
             _add_if_not_duplicate(resource_diff.resource_refs_only_right, i)
 
     return resource_diff
+
+
+@dataclasses.dataclass
+class OcmRepositoryConfig:
+    type: str
+
+
+@dataclasses.dataclass
+class OcmSoftwareResolverConfig:
+    repository: OcmRepositoryConfig
+    prefix: str
+    priority: int
+
+
+@dataclasses.dataclass
+class OcmSoftwareConfig:
+    type: str
+    aliases: typing.Optional[dict[str, dict]]
+    resolvers: list[OcmSoftwareResolverConfig]
+
+
+@dataclasses.dataclass
+class OcmLookupMapping:
+    ocm_repo_url: str
+    component_names: list[str]
+    use_for: oa.Privileges
+
+
+class OcmLookupMappingConfig:
+    def __init__(
+        self,
+        mappings: list[OcmLookupMapping],
+    ):
+        self.mappings = mappings
+        self._ro_lookups: list[tuple[typing.Callable[[str], bool], cm.OciRepositoryContext]] = []
+        self._rw_lookups: list[tuple[typing.Callable[[str], bool], cm.OciRepositoryContext]] = []
+
+        for mapping in mappings:
+            filter_func = reutil.re_filter(include_regexes=mapping.component_names)
+            ctx_repo = cm.OciRepositoryContext(baseUrl=mapping.ocm_repo_url)
+            filter_lookup = (filter_func, ctx_repo)
+
+            if mapping.use_for is oa.Privileges.READWRITE:
+                self._rw_lookups.append(filter_lookup)
+            self._ro_lookups.append(filter_lookup)
+
+    def apply_to(
+        self,
+        component_name,
+        use_for: str | oa.Privileges = 'readonly',
```

Comment:
nit: why not init w/ enum-value?

also: rather call those (required_)privileges?

Reply:
do we actually need to differentiate privileges / "use-for" at this point? I think the mapping might just yield _a_ oci-repository, regardless of intent